src/libsystemd/sd-network/sd-network.c

```diff
@@ -171,6 +171,17 @@ _public_ int sd_network_link_get_operational_state(int ifindex, char **state) {
         return network_link_get_string(ifindex, "OPER_STATE", state);
 }
 
+_public_ int sd_network_link_get_required_for_online(int ifindex) {
+        char* s;
+        int r;
+
+        r = network_link_get_string(ifindex, "REQUIRED_FOR_ONLINE", &s);
+        if (r < 0)
+            return r;
+
+        return parse_boolean(s);
```

Comment:
when the field doesn't exist, then this should be implicitly returned as "true", to maintain some level of compatibility.